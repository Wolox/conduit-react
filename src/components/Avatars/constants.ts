import Man1 from 'assets/avatars/Man1.png';
import Man2 from 'assets/avatars/Man2.png';
import Man3 from 'assets/avatars/Man3.png';
import Man4 from 'assets/avatars/Man4.png';
import Woman1 from 'assets/avatars/Woman1.png';
import Woman2 from 'assets/avatars/Woman2.png';
import Woman3 from 'assets/avatars/Woman3.png';
import Woman4 from 'assets/avatars/Woman4.png';
import { Avatar } from 'types/Avatar';

// export const AVATARS = [Man2, Man3, Man4, Woman1, Woman2, Woman3, Woman4];

export const AVATARS: Avatar[] = [
  { name: 'Man1', icon: Man1 },
  { name: 'Man2', icon: Man2 },
  { name: 'Man3', icon: Man3 },
  { name: 'Man4', icon: Man4 },
  { name: 'Woman1', icon: Woman1 },
  { name: 'Woman2', icon: Woman2 },
  { name: 'Woman3', icon: Woman3 },
  { name: 'Woman4', icon: Woman4 }
];
