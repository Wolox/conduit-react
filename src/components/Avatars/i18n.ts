import i18next from 'i18next';

i18next.addResources('en', 'Avatars', {
  title: 'Select your favorite avatar'
});

i18next.addResources('es', 'Avatars', {
  title: 'Selecciona tu avatar favorito'
});
