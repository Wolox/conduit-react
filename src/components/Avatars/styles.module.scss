@import 'scss/variables/_colors';
@import 'scss/fonts/_fonts';

.title {
  color: $green-fern;
  font-family: $titillium;
  font-size: 30px;
  font-weight: 600;
  line-height: 1;
  margin: 20px 0;
}

.content-image {
  margin: 20px 0;
  text-align: center;

  .img {
    border-radius: 100%;
    border: 2px solid $green-fern;
    cursor: pointer;
    height: auto;
    max-width: fit-content;
    width: 200px;
  }
}

.section-avatars {
  display: grid;
  grid-gap: 40px;
  grid-template-columns: repeat(auto-fill, 200px);
  justify-content: center;
  padding: 15px;
}

.img-avatar {
  background-color: $gray;
  border-radius: 100%;
  border: 2px solid $green-fern;
  height: auto;
  width: 100%;

  &:hover {
    background-color: $green-fern;
    cursor: pointer;
  }
}
