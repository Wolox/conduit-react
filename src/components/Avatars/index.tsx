import React, { useCallback, useState } from 'react';
import { useTranslation } from 'react-i18next';

import Modal from 'components/Modal';
import { getAvatar } from 'utils/avatarUtils';

import styles from './styles.module.scss';
import { AVATARS } from './constants';

interface Props {
  avatar?: string;
  changeAvatar: (avatar: string) => void;
}

function Avatars({ avatar, changeAvatar }: Props) {
  const [open, setOpen] = useState(false);
  const { t } = useTranslation('Avatars');

  const handleCloseModal = useCallback(() => {
    setOpen(false);
  }, []);

  const handleOpenModal = useCallback(() => setOpen(true), []);
  const handleClickAvatar = useCallback(
    (nameAvatarSelected) => {
      setOpen(false);
      changeAvatar(nameAvatarSelected);
    },
    [changeAvatar]
  );

  const { icon: currentIcon, name: currentName } = getAvatar(avatar || '');

  return (
    <div className={styles.contentImage}>
      <img src={currentIcon} alt={currentName} className={styles.img} onClick={handleOpenModal} />
      <Modal open={open} closeModal={handleCloseModal}>
        <h1 className={styles.title}>{t('title')}</h1>
        <div className={styles.sectionAvatars}>
          {AVATARS.map(({ icon, name }) => (
            <div key={name} onClick={() => handleClickAvatar(name)}>
              <img src={icon} alt={name} className={styles.imgAvatar} />
            </div>
          ))}
        </div>
      </Modal>
    </div>
  );
}

export default Avatars;
