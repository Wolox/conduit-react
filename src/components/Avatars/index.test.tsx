import { createMemoryHistory } from 'history';
import { ReactNode } from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Router } from 'react-router';

import Avatars from '.';

function Wrapped({ children }: { children: ReactNode }) {
  const queryClient = new QueryClient();
  const history = createMemoryHistory();
  return (
    <QueryClientProvider client={queryClient}>
      <Router history={history}>{children}</Router>
    </QueryClientProvider>
  );
}

describe('Component Avatars', () => {
  test('Renders component', () => {
    const { asFragment } = render(
      <Wrapped>
        <Avatars changeAvatar={jest.fn()} />
      </Wrapped>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('Display avatars in the modal', async () => {
    const changeAvatarMock = jest.fn();
    render(
      <Wrapped>
        <Avatars changeAvatar={changeAvatarMock} />
      </Wrapped>
    );
    const initialImagesLength = screen.getAllByRole('img').length;

    const imageProfile = screen.getByRole('img');
    userEvent.click(imageProfile);

    const totalImages = await waitFor(() => screen.findAllByRole('img'));
    expect(totalImages.length > initialImagesLength).toBeTruthy();
  });

  test('Change avatar selected', async () => {
    const changeAvatarMock = jest.fn();
    render(
      <Wrapped>
        <Avatars changeAvatar={changeAvatarMock} />
      </Wrapped>
    );
    const imagesProfile = screen.getByRole('img');
    userEvent.click(imagesProfile);

    const totalImages = await waitFor(() => screen.findAllByRole('img'));
    userEvent.click(totalImages[1]);
    expect(changeAvatarMock).toBeCalled();
  });
});
