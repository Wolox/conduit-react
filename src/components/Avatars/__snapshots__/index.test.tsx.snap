// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Avatars Renders component 1`] = `
<DocumentFragment>
  <div
    class="contentImage"
  >
    <img
      alt="Default Image"
      class="img"
      src="user-placeholder.jpeg"
    />
  </div>
</DocumentFragment>
`;
